nautobot/ipam/signals.py

```diff
@@ -0,0 +1,44 @@
+from django.core.exceptions import ValidationError
+from django.db.models.signals import pre_save, m2m_changed
+from django.dispatch import receiver
+
+from nautobot.ipam.models import Prefix, VRF
+
+
+@receiver(pre_save, sender=Prefix)
+@receiver(pre_save, sender=VRF)
+@receiver(pre_save, sender=VRF.devices.through)
+@receiver(pre_save, sender=VRF.prefixes.through)
+def ipam_object_saved(sender, instance, raw=False, **kwargs):
+    """
+    Forcibly call `full_clean()` when a new IPAM object is manually saved to prevent creation of
+    invalid objects.
+    """
+    if raw:
+        return
+    instance.full_clean()
```

Comment:
This is going to create some noise, as it's causing a bunch of tests to blow up everywhere. See: https://github.com/nautobot/nautobot/pull/3378/files#diff-00a7372c809230ca828aa70f90352848b6654faadb78e4762a8f55ee6c59ce4fR584-R585

Reply:
That is correct. This is there for now to force model validation in lieu of having formsets for m2m intermediate relationships for the prototype. I don't think it's worth creating the formset code right now if we're just going to rip it out and replace it for the v2 UI.

Reply:
I disabled it for `Prefix` and `VRF`.